src/main/kotlin/koren/proj/puzzlesolver/model/searchAlgo/DFS.kt

```diff
@@ -1,13 +1,51 @@
 package koren.proj.puzzlesolver.model.searchAlgo
 
 import koren.proj.puzzlesolver.model.puzzle.AbstractPuzzle
+import java.util.*
+import kotlin.collections.HashMap
+import kotlin.collections.HashSet
 
 class DFS : GenericSearchInterface {
 
-    override var parentMap: HashMap<AbstractPuzzle, AbstractPuzzle> = HashMap()
-    override var visitedSet: HashSet<AbstractPuzzle> = HashSet()
+    private val parentMap: HashMap<AbstractPuzzle, AbstractPuzzle> = HashMap()
+    private val visitedSet: HashSet<AbstractPuzzle> = HashSet()
 
-    override fun search(initial_puzzle: AbstractPuzzle, goal: AbstractPuzzle) {
-        TODO("Not yet implemented")
+    override fun search(initialPuzzle: AbstractPuzzle, goal: AbstractPuzzle): Collection<AbstractPuzzle> {
```

Comment:
Change from collection

Reply:
Add a test which uses `search()` twice on same input and receives correct result after the second time